### DISCUSSION TITLE
Prop `className` did not match

### DISCUSSION THREAD

**Discussion by sleepinzombie** [vercel/next.js#49404] [Help]
### Summary

Hi all, I am having issue with my recent Next.js installation locally whenever I am serving the app.

Before I explain further, the dependencies I am currently using are:
```
"dependencies": {
    "@emotion/react": "^11.11.0",
    "@emotion/styled": "^11.11.0",
    "@mui/material": "^5.12.3",
    "@types/node": "20.1.0",
    "@types/react": "18.2.6",
    "@types/react-dom": "18.2.4",
    "eslint": "8.40.0",
    "eslint-config-next": "13.4.1",
    "next": "13.4.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "typescript": "5.0.4"
  },
```

Whenever I am running the app locally, I get this error:
<img width="832" alt="image" src="https://user-images.githubusercontent.com/39788411/236686338-b6b090a8-72ea-400a-bb55-cfb2f3bca5a9.png">

My Next config are as follows:
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: false,
  experimental: {
    appDir: true,
  },
  compiler: {
    styledComponents: true,
  },
};

module.exports = nextConfig;
```

And the code itself is:
```
import { Inter } from 'next/font/google';
import styles from './styles.module.css';

const inter = Inter({ subsets: ['latin'] });

export const metadata = {
  title: 'Signup | Projector',
  description: 'Generated by Next.js',
};

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en" className={styles.fullHeight}>
      <body className={inter.className + ' ' + styles.fullHeight}>
        {children}
      </body>
    </html>
  );
}

```

Can someone help me on this, please? I'm stuck for hours.

Thanking you all in advance.

**Comment by icyJoseph** [comment]
Have you followed these steps? https://nextjs.org/docs/app/building-your-application/styling/css-in-js#styled-components

Your root layout page would suggest you haven't. 

This error happens because the server generated data is never collected. If you have ever used Styled Components, you know, we had to collect the styles from the application inside _document. 

In Server Components (app router), you have to do something similar, through a component registry.

**Reply by ivangrimaldo** [reply]
if you take a look of your link, you can see the same configuration in next.config.js

Im facing the same problem and this occurs when the body element get the className from inter class when you use `import { Inter } from 'next/font/google';` 

the problem always originated in `layout.tsx`


must of the answers point the same configuration of the compiler and the `use client` property of the files.

by he way, cant use `use client` in file layout.ts

![image](https://github.com/vercel/next.js/assets/129094318/e0fdc9aa-dc90-4b65-b22e-4e88218aff8d)
